File: home_assignments/java/TicketSeller/TicketSellerModel/src/main/java/ticketseller/model/validators/interfaces/IValidator.java
package ticketseller.model.validators.interfaces;

import ticketseller.model.entities.Identifiable;
import ticketseller.model.validators.ValidationException;

public interface IValidator<ID, E extends Identifiable<ID>> {
    /**
     * validates an entity
     *
     * @param entity said entity
     * @throws ValidationException if entity is invalid
     */
    void validate(E entity) throws ValidationException;
}

File: home_assignments/csharp/TicketSeller/model/validators/interfaces/IValidator.cs
using model.entities;

namespace model.validators.interfaces;

public interface IValidator<TId, TEntity> where TEntity : Entity<TId>
{
    
    /// <summary>
    /// validates an entity
    /// </summary>
    /// <param name="entity">said entity</param>
    /// <exception cref="ValidationException">if entity is invalid</exception>
    void Validate(TEntity entity);
}